idaes/core/solvers/petsc.py

```diff
@@ -477,10 +478,17 @@ def petsc_dae_by_time_element(
             Pyomo model.
         previous_trajectory: (PetscTrajectory) Trajectory from previous integration
             of this model. New results will be appended to this trajectory object.
+        snes_options (dict): [DEPRECATED in favor of initial_solver_options] nonlinear equation solver options
 
     Returns (PetscDAEResults):
         See PetscDAEResults documentation for more information.
     """
+    if snes_options is not None and initial_solver_options is not None:
+        raise RuntimeError(
+            "Both (deprecated) snes_options and initial_solver_options keyword arguments were specified. "
+            "Please specify your initial solver options using only the initial_solver_options argument."
+        )
```

Comment:
For clarity to the reader, I would be inclined to move the deprecation warning to immediately after this. This makes it clear why you can;t have both options, and also means that from this point on you don;t have to worry about the deprecation option at all (at the moment, there is the possibility of something happening to the two options between here and when the deprecation is checked).